src/main/java/world/bentobox/bentobox/managers/IslandDeleteManager.java

```diff
@@ -0,0 +1,84 @@
+package world.bentobox.bentobox.managers;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+
+import world.bentobox.bentobox.BentoBox;
+import world.bentobox.bentobox.api.events.BentoBoxReadyEvent;
+import world.bentobox.bentobox.api.events.island.IslandEvent.IslandDeleteChunksEvent;
+import world.bentobox.bentobox.api.events.island.IslandEvent.IslandDeletedEvent;
+import world.bentobox.bentobox.database.Database;
+import world.bentobox.bentobox.database.objects.DeletedIslandDO;
+import world.bentobox.bentobox.util.DeleteIslandChunks;
+import world.bentobox.bentobox.util.Util;
+
+/**
+ * Listens for island deletions and adds them to the database. Removes them when the island is deleted.
+ * @author tastybento
+ *
+ */
+public class IslandDeleteManager implements Listener {
```

Comment:
Not the best name as well, I think `IslandDeletionManager` would be more appropriate.